community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/AwaitIndexProcedureTest.java

```diff
@@ -0,0 +1,339 @@
+        new Thread( () ->
+        {
+            try
+            {
+                procedure.execute( null, null, timeout, timeoutUnits );
+            }
+            catch ( ProcedureException e )
+            {
+                exception.set( e );
+            }
+        } ).start();
+
+        assertEventually( "Procedure did not time out", exception::get, not( nullValue() ), timeout * 2, timeoutUnits );
+    }
+
+    private class StubKernelTransaction implements KernelTransaction
+    {
+        private final ReadOperations readOperations;
+
+        public StubKernelTransaction( ReadOperations readOperations )
+        {
+            this.readOperations = readOperations;
+        }
+
+        @Override
+        public Statement acquireStatement()
+        {
+            return new StubStatement( readOperations );
+        }
+
+        @Override
+        public void success()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public void failure()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public long closeTransaction() throws TransactionFailureException
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public boolean isOpen()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public AccessMode mode()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public Status getReasonIfTerminated()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public void markForTermination( Status reason )
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public long lastTransactionTimestampWhenStarted()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public long lastTransactionIdWhenStarted()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public long localStartTime()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public void registerCloseListener( CloseListener listener )
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public Type transactionType()
+        {
+            throw new UnsupportedOperationException( "not implemented" );
+        }
+
+        @Override
+        public Revertable restrict( AccessMode mode )
+        {
+            throw new UnsupportedOperationException( "not implemented" );
```

Comment:
Well, it could be. But I'm not a big fan of using mocks for stubbing where it's straightforward to implement an interface.